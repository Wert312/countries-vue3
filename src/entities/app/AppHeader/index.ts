import AppHeader from "@/entities/app/AppHeader/ui/index.vue";

export default AppHeader;
