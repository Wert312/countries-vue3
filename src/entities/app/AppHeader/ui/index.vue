<template>
  <div class="header">
    <div class="header__inner-content">
      <div class="header__link" @click="scrollTop">
        <RouterLink
          tag="div"
          :to="{ name: 'CountryGridPage' }"
          class="header__title"
          >Where in the world?</RouterLink
        >
      </div>
      <div @click="toggleCurrentMode" class="header__switch-mode">
        toggle theme
      </div>
    </div>
  </div>
</template>

<script setup lang="ts">
import { onMounted, ref } from "vue";

onMounted(() => {
  setCurrentMode();
});

let currentMode = ref(localStorage.getItem("theme"));

function setCurrentMode() {
  document.documentElement.setAttribute("data-theme", `${currentMode.value}`);
}

function scrollTop() {
  window.scrollTo(0, 0);
}

function toggleCurrentMode() {
  if (currentMode.value === "dark") {
    currentMode.value = "light";
  } else {
    currentMode.value = "dark";
  }
  document.documentElement.setAttribute("data-theme", `${currentMode.value}`);
  localStorage.setItem("theme", `${currentMode.value}`);
}
</script>

<style scoped lang="scss">
.header {
  color: var(--app-text);
  background-color: var(--color-light);
  display: flex;
  position: fixed;
  z-index: 1000;
  align-items: center;
  width: 100%;
  height: 80px;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 15%);

  &__inner-content {
    width: 100%;
    padding: 0 16px;
    margin: 0 auto;
    max-width: 1140px;
    display: flex;
    align-items: center;
    justify-content: space-between;
  }

  &__title {
    font-style: normal;
    font-weight: 800;
    color: var(--app-text);
    text-decoration: none;
    font-size: 14px;
    line-height: 20px;
    cursor: pointer;

    @include for-tablet {
      font-size: 24px;
      line-height: 33px;
    }
  }

  &__switch-mode {
    font-style: normal;
    font-weight: 600;
    font-size: 12px;
    line-height: 16px;
    cursor: pointer;

    @include for-tablet {
      font-size: 16px;
      line-height: 22px;
    }
  }
}
</style>
