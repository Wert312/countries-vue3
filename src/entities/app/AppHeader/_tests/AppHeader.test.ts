import { mount } from "@vue/test-utils";
import { expect, it } from "vitest";
import AppHeader from "../index";

describe("AppHeader tests", () => {
  it("should render", () => {
    const wrapper = mount(AppHeader);
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
