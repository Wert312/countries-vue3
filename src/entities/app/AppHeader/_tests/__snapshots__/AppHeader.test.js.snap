// Vitest Snapshot v1

exports[`AppError tests > should render 1`] = `
VueWrapper {
  "__app": {
    "_component": {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="header"
        data-v-c6318512=""
      >
        <div
          class="header__inner-content"
          data-v-c6318512=""
        >
          <div
            class="header__link"
            data-v-c6318512=""
          >
            <routerlink
              class="header__title"
              data-v-c6318512=""
              tag="div"
              to="[object Object]"
            >
              Where in the world?
            </routerlink>
          </div>
          <div
            class="header__switch-mode"
            data-v-c6318512=""
          >
             toggle theme 
          </div>
        </div>
      </div>
    </div>,
    "_context": {
      "app": [Circular],
      "components": {
        "transition": {
          "name": "transition",
        },
        "transition-group": {
          "name": "transition-group",
        },
      },
      "config": {
        "compilerOptions": {},
        "errorHandler": undefined,
        "globalProperties": {},
        "isNativeTag": [Function],
        "optionMergeStrategies": {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": {},
      "emitsCache": WeakMap {},
      "mixins": [
        {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": {},
      "reload": [Function],
    },
    "_instance": {
      "a": null,
      "accessCache": {},
      "appContext": {
        "app": [Circular],
        "components": {
          "transition": {
            "name": "transition",
          },
          "transition-group": {
            "name": "transition-group",
          },
        },
        "config": {
          "compilerOptions": {},
          "errorHandler": undefined,
          "globalProperties": {},
          "isNativeTag": [Function],
          "optionMergeStrategies": {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": {},
        "emitsCache": WeakMap {},
        "mixins": [
          {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": {},
      "da": null,
      "data": {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": {},
      "propsDefaults": {},
      "propsOptions": [
        {},
        [],
      ],
      "provides": {},
      "proxy": {},
      "refs": {
        "VTU_COMPONENT": {
          "hasOwnProperty": [Function],
        },
      },
      "render": [Function],
      "renderCache": [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": [],
        "effects": [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": {},
      "slots": {},
      "sp": null,
      "subTree": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": {
          "a": null,
          "accessCache": {},
          "appContext": {
            "app": [Circular],
            "components": {
              "transition": {
                "name": "transition",
              },
              "transition-group": {
                "name": "transition-group",
              },
            },
            "config": {
              "compilerOptions": {},
              "errorHandler": undefined,
              "globalProperties": {},
              "isNativeTag": [Function],
              "optionMergeStrategies": {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": {},
            "emitsCache": WeakMap {},
            "mixins": [
              {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": {},
          "ctx": {},
          "da": null,
          "data": {},
          "devtoolsRawSetupState": {
            "currentMode": RefImpl {
              "__v_isRef": true,
              "__v_isShallow": false,
              "_rawValue": null,
              "_value": null,
              "dep": undefined,
            },
            "scrollTop": [Function],
            "setCurrentMode": [Function],
            "toggleCurrentMode": [Function],
          },
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": [],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": {
            "hasOwnProperty": [Function],
          },
          "exposed": {
            "hasOwnProperty": [Function],
          },
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": [
            [Function],
          ],
          "next": null,
          "parent": [Circular],
          "props": {},
          "propsDefaults": {},
          "propsOptions": [
            {},
            [],
          ],
          "provides": {},
          "proxy": {},
          "refs": {},
          "render": [Function],
          "renderCache": [],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": [],
            "effects": [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": [],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": {
            "attrs": {},
            "emit": [Function],
            "expose": [Function],
            "slots": {},
          },
          "setupState": {
            "currentMode": null,
            "scrollTop": [Function],
            "setCurrentMode": [Function],
            "toggleCurrentMode": [Function],
          },
          "slots": {},
          "sp": null,
          "subTree": {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": null,
            "appContext": null,
            "children": [
              {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": [
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": [
                      {
                        "__v_isVNode": true,
                        "__v_skip": true,
                        "anchor": null,
                        "appContext": null,
                        "children": [
                          {
                            "__v_isVNode": true,
                            "__v_skip": true,
                            "anchor": null,
                            "appContext": null,
                            "children": "Where in the world?",
                            "component": null,
                            "dirs": null,
                            "dynamicChildren": null,
                            "dynamicProps": null,
                            "el": Where in the world?,
                            "key": null,
                            "patchFlag": 0,
                            "props": null,
                            "ref": null,
                            "scopeId": null,
                            "shapeFlag": 8,
                            "slotScopeIds": null,
                            "ssContent": null,
                            "ssFallback": null,
                            "staticCount": 0,
                            "suspense": null,
                            "target": null,
                            "targetAnchor": null,
                            "transition": null,
                            "type": Symbol(Text),
                          },
                        ],
                        "component": null,
                        "dirs": null,
                        "dynamicChildren": null,
                        "dynamicProps": null,
                        "el": <routerlink
                          class="header__title"
                          data-v-c6318512=""
                          tag="div"
                          to="[object Object]"
                        >
                          Where in the world?
                        </routerlink>,
                        "key": null,
                        "patchFlag": 0,
                        "props": {
                          "class": "header__title",
                          "tag": "div",
                          "to": {
                            "name": "CountryGridPage",
                          },
                        },
                        "ref": null,
                        "scopeId": "data-v-c6318512",
                        "shapeFlag": 17,
                        "slotScopeIds": null,
                        "ssContent": null,
                        "ssFallback": null,
                        "staticCount": 0,
                        "suspense": null,
                        "target": null,
                        "targetAnchor": null,
                        "transition": null,
                        "type": "RouterLink",
                      },
                    ],
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <div
                      class="header__link"
                      data-v-c6318512=""
                    >
                      <routerlink
                        class="header__title"
                        data-v-c6318512=""
                        tag="div"
                        to="[object Object]"
                      >
                        Where in the world?
                      </routerlink>
                    </div>,
                    "key": null,
                    "patchFlag": 0,
                    "props": {
                      "class": "header__link",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "scopeId": "data-v-c6318512",
                    "shapeFlag": 17,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "div",
                  },
                  {
                    "__v_isVNode": true,
                    "__v_skip": true,
                    "anchor": null,
                    "appContext": null,
                    "children": " toggle theme ",
                    "component": null,
                    "dirs": null,
                    "dynamicChildren": null,
                    "dynamicProps": null,
                    "el": <div
                      class="header__switch-mode"
                      data-v-c6318512=""
                    >
                       toggle theme 
                    </div>,
                    "key": null,
                    "patchFlag": 0,
                    "props": {
                      "class": "header__switch-mode",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "scopeId": "data-v-c6318512",
                    "shapeFlag": 9,
                    "slotScopeIds": null,
                    "ssContent": null,
                    "ssFallback": null,
                    "staticCount": 0,
                    "suspense": null,
                    "target": null,
                    "targetAnchor": null,
                    "transition": null,
                    "type": "div",
                  },
                ],
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <div
                  class="header__inner-content"
                  data-v-c6318512=""
                >
                  <div
                    class="header__link"
                    data-v-c6318512=""
                  >
                    <routerlink
                      class="header__title"
                      data-v-c6318512=""
                      tag="div"
                      to="[object Object]"
                    >
                      Where in the world?
                    </routerlink>
                  </div>
                  <div
                    class="header__switch-mode"
                    data-v-c6318512=""
                  >
                     toggle theme 
                  </div>
                </div>,
                "key": null,
                "patchFlag": 0,
                "props": {
                  "class": "header__inner-content",
                },
                "ref": null,
                "scopeId": "data-v-c6318512",
                "shapeFlag": 17,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "div",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": [],
            "dynamicProps": null,
            "el": <div
              class="header"
              data-v-c6318512=""
            >
              <div
                class="header__inner-content"
                data-v-c6318512=""
              >
                <div
                  class="header__link"
                  data-v-c6318512=""
                >
                  <routerlink
                    class="header__title"
                    data-v-c6318512=""
                    tag="div"
                    to="[object Object]"
                  >
                    Where in the world?
                  </routerlink>
                </div>
                <div
                  class="header__switch-mode"
                  data-v-c6318512=""
                >
                   toggle theme 
                </div>
              </div>
            </div>,
            "key": null,
            "patchFlag": 0,
            "props": {
              "class": "header",
            },
            "ref": null,
            "scopeId": "data-v-c6318512",
            "shapeFlag": 17,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": "div",
          },
          "suspense": null,
          "suspenseId": 0,
          "type": {
            "__file": "C:/Users/averkhusha/WebstormProjects/countries-vue3/src/entities/app/AppHeader/ui/index.vue",
            "__name": "index",
            "__scopeId": "data-v-c6318512",
            "components": {},
            "render": [Function],
            "setup": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="header"
          data-v-c6318512=""
        >
          <div
            class="header__inner-content"
            data-v-c6318512=""
          >
            <div
              class="header__link"
              data-v-c6318512=""
            >
              <routerlink
                class="header__title"
                data-v-c6318512=""
                tag="div"
                to="[object Object]"
              >
                Where in the world?
              </routerlink>
            </div>
            <div
              class="header__switch-mode"
              data-v-c6318512=""
            >
               toggle theme 
            </div>
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": {
          "ref": "VTU_COMPONENT",
        },
        "ref": {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "__file": "C:/Users/averkhusha/WebstormProjects/countries-vue3/src/entities/app/AppHeader/ui/index.vue",
          "__name": "index",
          "__scopeId": "data-v-c6318512",
          "components": {},
          "render": [Function],
          "setup": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": {
          "app": [Circular],
          "components": {
            "transition": {
              "name": "transition",
            },
            "transition-group": {
              "name": "transition-group",
            },
          },
          "config": {
            "compilerOptions": {},
            "errorHandler": undefined,
            "globalProperties": {},
            "isNativeTag": [Function],
            "optionMergeStrategies": {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": {},
          "emitsCache": WeakMap {},
          "mixins": [
            {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": <div
          class="header"
          data-v-c6318512=""
        >
          <div
            class="header__inner-content"
            data-v-c6318512=""
          >
            <div
              class="header__link"
              data-v-c6318512=""
            >
              <routerlink
                class="header__title"
                data-v-c6318512=""
                tag="div"
                to="[object Object]"
              >
                Where in the world?
              </routerlink>
            </div>
            <div
              class="header__switch-mode"
              data-v-c6318512=""
            >
               toggle theme 
            </div>
          </div>
        </div>,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": {
      "compilerOptions": {},
      "errorHandler": undefined,
      "globalProperties": {},
      "isNativeTag": [Function],
      "optionMergeStrategies": {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.38",
  },
  "__setProps": [Function],
  "componentVM": {},
  "isDisabled": [Function],
  "rootVM": {},
  "wrapperElement": <div
    class="header"
    data-v-c6318512=""
  >
    <div
      class="header__inner-content"
      data-v-c6318512=""
    >
      <div
        class="header__link"
        data-v-c6318512=""
      >
        <routerlink
          class="header__title"
          data-v-c6318512=""
          tag="div"
          to="[object Object]"
        >
          Where in the world?
        </routerlink>
      </div>
      <div
        class="header__switch-mode"
        data-v-c6318512=""
      >
         toggle theme 
      </div>
    </div>
  </div>,
}
`;
